tower-http/src/metrics.rs

```diff
@@ -0,0 +1,337 @@
+use crate::{common::*, GetTraceStatus, GetTraceStatusFromHttpStatus, LatencyUnit, TraceStatus};
+use http::{Method, Version};
+use metrics::{gauge, histogram, increment_counter, SharedString};
```

Comment:
I'd be curious if we could implement this with a trait and optionally pull in metrics? That way I can implement it for some of our internal metrics stuff at work :)

Reply:
Similarly, we don't use the metrics library. I think it would be unfortunate if a general `tower-http` crate set of utilities pulled in specific external implementations like this. This seems better suited to its own crate, imo.